File: hadoop/src/ContadorDePalavrasReducer.java
import org.apache.hadoop.io.IntWritable;
import org.apache.hadoop.io.Text;
import org.apache.hadoop.mapreduce.Reducer;

import java.io.IOException;

public class ContadorDePalavrasReducer extends Reducer<Text, IntWritable, IntWritable, Text> {

    @Override
    protected void reduce(Text key, Iterable<IntWritable> values, Context context) throws IOException, InterruptedException {
        int sum = 0;

        // Soma todas as ocorrências da palavra
        for (IntWritable value : values) {
            sum += value.get();
        }

        // Emite a palavra com o total de ocorrências
        context.write(new IntWritable(sum), key);
    }
}

File: hadoop/src/ContadorDePalavrasReducer.scala
import org.apache.hadoop.io.{IntWritable, Text}
import org.apache.hadoop.mapreduce.Reducer
import scala.jdk.CollectionConverters._ 

class ContadorDePalavrasReducer extends Reducer[Text, IntWritable, IntWritable, Text] {
  override def reduce(key: Text, values: java.lang.Iterable[IntWritable], context: Reducer[Text, IntWritable, IntWritable, Text]#Context): Unit = {
    // Converte o java.util.Iterator em um Scala Iterator usando asScala
    val sum = values.iterator().asScala.foldLeft(0)(_ + _.get)
    context.write(new IntWritable(sum), key)
  }
}
